lib/aggregate-translations.js

```diff
@@ -76,6 +81,8 @@ const aggregatedTranslations = (options) => {
 
   // Write intl and translations loaders for the specified locales
   writeI18nLoaders(locales, fileSystem, outputDirectory, format);
+
+  return locales;
```

Comment:
If a failure occurs during aggregation, will this still return the full list aggregated on or does the whole things stop? I don't remember how it's been setup to handle failures.

Reply:
It depends on the failure, if a code block fails on a syntax error or file read the code will stop execution. If a non-terra supported locale is detected the loaders and translation files will still be generated for the locale and the full list will be returned, but it does log out warnings.

Example of providing a non-terra locale. In this case a locale named "bad-locale".

```
WARNING: bad-locale is NOT a Terra supported locale. Creating an aggregate translation file for bad-locale, but be sure to include the appropriate translations messages for each terra component used in your application otherwise the translations will fail and the fallback will be displayed.
WARNING: bad-locale is NOT a Terra supported locale. Creating a translation and intl loader for bad-locale, but be sure the lanaguage specified is supported by intl, otherwise no locale-date will exist and the import will result in an error.
```

Because the loaders are still created webpack will fail if the locale isn't successfully aggregated/supported:

```
ERROR in ./aggregated-translations/intlLoaders.js
Module not found: Error: Can't resolve 'intl/locale-data/jsonp/bad-locale.js' in '/Users/se028266/workspaces/react-on-rails/terra-application/aggregated-translations'
 @ ./aggregated-translations/intlLoaders.js 11:11-58
 @ ./node_modules/terra-i18n/lib/intlLoaders.js
 @ ./node_modules/terra-i18n/lib/i18nLoader.js
 @ ./node_modules/terra-i18n/lib/I18n.js
 @ ./node_modules/terra-base/lib/Base.js
 @ ./src/application-base/ApplicationBase.jsx
 @ ./node_modules/terra-dev-site/lib/TerraDevSite.js

ERROR in ./aggregated-translations/bad-locale.js
Module not found: Error: Can't resolve 'react-intl/locale-data/bad' in '/Users/se028266/workspaces/react-on-rails/terra-application/aggregated-translations'
 @ ./aggregated-translations/bad-locale.js 10:11-48
 @ ./aggregated-translations/translationsLoaders.js
 @ ./node_modules/terra-i18n/lib/translationsLoaders.js
 @ ./node_modules/terra-i18n/lib/i18nLoader.js
 @ ./node_modules/terra-i18n/lib/I18n.js
 @ ./node_modules/terra-base/lib/Base.js
 @ ./src/application-base/ApplicationBase.jsx
 @ ./node_modules/terra-dev-site/lib/TerraDevSite.js
```